File: csharp_body_deref/Scripts/Settings/WeatherSettings.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Settings.WeatherSettings
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Settings/WeatherSettings.c
// Prior real C# source (older reference): Assets/Scripts/Settings/InGame/WeatherSettings.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;

namespace Settings;

[Token(Token = "0x2000098")]
internal class WeatherSettings : PresetSettingsContainer
{
	[Token(Token = "0x400037D")]
	[FieldOffset(Offset = "0x20")]
	public SetSettingsContainer<WeatherSet> WeatherSets;

	[Token(Token = "0x1700008E")]
	protected override string FileName
	{
		[Token(Token = "0x6000338")]
		[Address(RVA = "0x3F8CD60", Offset = "0x3F8CD60", VA = "0x3F8CD60", Slot = "14")]
		get
		{
			return null;
		}
	}

	[Token(Token = "0x6000339")]
	[Address(RVA = "0x3F8CD90", Offset = "0x3F8CD90", VA = "0x3F8CD90")]
	public WeatherSettings()
	{
	}
}

File: cpp/Settings/WeatherSettings.c
// Type: Settings.WeatherSettings
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Settings/WeatherSettings.cs
// Prior real C# source (older reference): Assets/Scripts/Settings/InGame/WeatherSettings.cs
// --------------------------------

// Settings.WeatherSettings$$get_FileName
// il2cpp: System_String_o* Settings_WeatherSettings__get_FileName (Settings_WeatherSettings_o* __this, const MethodInfo* method);
// 0x408cd60

System_String_o *
Settings_WeatherSettings__get_FileName(Settings_WeatherSettings_o *__this,MethodInfo *method)

{
  if (g_data_057ac3fd == '\0') {
    il2cpp_runtime_helper_023445d0(&"Weather.json");
    g_data_057ac3fd = '\x01';
  }
  return "Weather.json";
}


// Settings.WeatherSettings$$.ctor
// il2cpp: void Settings_WeatherSettings___ctor (Settings_WeatherSettings_o* __this, const MethodInfo* method);
// 0x408cd90

void Settings_WeatherSettings___ctor(Settings_WeatherSettings_o *__this,MethodInfo *method)

{
  Settings_SetSettingsContainer_WeatherSet__o *__this_00;
  
  if (g_data_057ac3fe == '\0') {
    il2cpp_runtime_helper_023445d0(&MethodInfo_SetSettingsContainer_1_WeatherSet);
    il2cpp_runtime_helper_023445d0(&TypeInfo_SetSettingsContainer_WeatherSet);
    g_data_057ac3fe = '\x01';
  }
  __this_00 = (Settings_SetSettingsContainer_WeatherSet__o *)il2cpp_runtime_helper_023052d0(TypeInfo_SetSettingsContainer_WeatherSet);
  Settings_SetSettingsContainer_object____ctor((Settings_SetSettingsContainer_T__o *)__this_00,MethodInfo_SetSettingsContainer_1_WeatherSet);
  (__this->fields).WeatherSets = __this_00;
  il2cpp_runtime_helper_022b4080(&(__this->fields).WeatherSets);
  Settings_PresetSettingsContainer___ctor((Settings_PresetSettingsContainer_o *)__this,(MethodInfo *)0x0);
  return;
}


